### ISSUE TITLE
New qTox tray icon has no way to restore on Plasma 5

### ISSUE THREAD

**Issue by 3vi1** [qTox/qTox#1065]
With the recent changes (not sure if Plasma 5, or qTox), I can now see the qTox icon in the plasma desktop system notification area, instead of in my legacy stalone tray.  That's good.

However, the new qTox icon, when left or double-clicked does nothing.  Right-clicking allows me to change the status or quit it...  but there is no option to restore the window.  That's bad.

Also, the icon is being drawn as a white sheet of paper with a question mark in it, which I am guessing means it did not provide a suitably sized icon in the format that plasma expects.

**Comment by agilob** [comment]
I've tried it yesterday and can't reproduce any of these(?)

**Comment by 3vi1** [comment]
![menu_009](https://cloud.githubusercontent.com/assets/628806/6072373/abd6fa24-ad66-11e4-9b7c-f02983da6ee9.png)

The page icon is qTox, and doubleclicking it does nothing.  This is with Ubuntu 15.04 and the latest plasma-desktop packages installed.  All other tray icons work fine.

**Comment by dubslow** [comment]
Could it be an upstream bug?

**Comment by 3vi1** [comment]
Anything's possible, but given that this is the only app demonstrating the symptoms, and that I'm compiling the latest qTox from git, I'm not sure how.

**Comment by dubslow** [comment]
Do you know how many other Qt 5 apps you have on your system?

**Comment by agilob** [comment]
I tested it on Arch with plasma 5.2.0-2 with all additonal tray support:

> Plasma 5 uses a new specification to display items in the 'system tray' area called Status Notifier. To allow applications using the old xembed specification to display items the packages (gtk-2)libappindicator-gtk2, (gtk-3)libappindicator-gtk3 or (qt4)sni-qt, lib32-sni-qt(for 32bit based applications like Skype) may be required. See "Where are my systray icons?" for more information. 

and it worked ok, images were a bit blury, but no problem with tray icon.

**Comment by 3vi1** [comment]
Has qTox not moved to using the current notifier specification?  If not, I suppose it could be a problem with the sni-qt packages in Vivid.  I've had all the legacy support packages (including sni-qt:i386) installed for a long time to support things like Skype and KeePass2 (which both work fine).

In answer to dubslow, I'm not sure how many other Qt5 apps I have.  I know I have some, like the current QtCreator, but the apps that I'm sure are Qt5 don't implement any notifiers.  It looks like the version of Amarok packaged in Vivid is still built against Qt4.

**Comment by tux3** [comment]
> Plasma 5 uses a new specification to display items in the 'system tray' area called Status Notifier.

It's like every DE is trying really hard to disregard every standard and create a brand new system tray API that isn't compatible with anything else.

**Comment by agilob** [comment]
https://xkcd.com/927/

**Comment by agilob** [comment]
Done?

**Comment by tux3** [comment]
Yep.

**Comment by 3vi1** [comment]
In recent weeks, rebuilding from source, I've noticed that the tray behavior of qTox has changed.

Unfortunately, the change is that it no longer creates a tray icon at all.  Toggling it on/off in the options has no effect.

I notice this when running from the command line:

```
SystemTrayIcon: Detected a KDE5 session, disabling the icon
Core: Starting up
```

**Comment by tux3** [comment]
@3vi1 please make sure you have the very latest version, the KDE5 StatusNotifier icon should now be enabled by default.

**Comment by agilob** [comment]
@3vi1 do you have sni-qt?
https://github.com/owncloud/client/issues/2023

**Comment by 3vi1** [comment]
Yes, I have the latest versions of sni-qt installed from the 15.04 repos:

```
4:16:39 evil@clevo [~/src/qtox]> dpkg -l | grep '^ii' | grep sni-qt
ii  sni-qt:amd64                                                0.2.6-0ubuntu1                                  amd64        indicator support for Qt
ii  sni-qt:i386                                                 0.2.6-0ubuntu1                                  i386         indicator support for Qt
```

**Comment by tux3** [comment]
Note that you don't actually need sni-qt for qTox on KDE5, since we follow the Status Notifier specification instead.

**Comment by 3vi1** [comment]
I've had sni-qt installed for a long time, for Skype.  (which shows its indicator correctly)

**Comment by 3vi1** [comment]
Is it possible there's some qt5 lib I'm missing to build it with tray support that simple_make doesn't complain about?

**Comment by tux3** [comment]
Again, please make sure you have the latest version of qTox.
As long as you don't pass ENABLE_SYSTRAY_STATUSNOTIFIER_BACKEND=NO to qmake, support for KDE5 tray icon will be compiled in.

**Comment by agilob** [comment]
Should it be compiled with Qt 5.4 or 5.5 then?

**Comment by tux3** [comment]
Either should work.

**Comment by 3vi1** [comment]
I just did a git stash; git stash drop; git pull; like I do every few days and re-built it and....

YOU'VE FIXED IT!  

I thought I did a git pull earlier, but maybe I didn't do the stash/drop that seems to be necessary after every run of simple_make.sh.

Great job!  The icon looks correct, and has the correct behavior when clicked (minimizing/restoring app).  Thanks for taking the time to fix that.  This can be closed now.

![tooltip_016](https://cloud.githubusercontent.com/assets/628806/6401611/589f3c42-bdc7-11e4-96e4-414120517474.png)
